themes/bootstrap3/templates/layout/layout.phtml

```diff
@@ -107,6 +107,12 @@
     ?>
     <?=$this->cookieConsent()->render()?>
     <?=$this->headScript() ?>
+    <?php if (isset($lightboxParent)): ?>
+      <script>window.lightboxParent = "<?= urlencode($lightboxParent) ?>";</script>
```

Comment:
Unless there's a really good reason to store these in window, I'd follow the existing practice of using the VuFind object for state (see `$appendScripts` right above). Add the scripts to `$appendScripts`, otherwise any CSP will block them since they won't get a nonce.

I'd also prefer for the `lightboxParent` and `lightboxChild` to not be urlencoded when they're not part of a URL. Otherwise the way they're used later on may be confusing, i.e. it may look like they're missing proper encoding when added to a URL. Plus you wouldn't need the `decodeURIComponent` and handling of spaces as pluses. You could use e.g. `json_encode((string)$lightboxParent)` to ensure they're safe in the JS code.

Reply:
I'll second the suggestion that it's probably more clear to encode at the time we need encoding, instead of in advance. Otherwise, it can become confusing, since it requires looking in multiple parts of the code to figure out whether or not encoding has already been applied. (And if there were a strong reason to encode in advance, I would reflect the encoding in the variable name, like lightboxParentUrlEncoded, to make this really clear... but obviously not encoding seems simpler than weird variable names in most situations).

Reply:
Thank you for the hint. I moved both urls to the VuFind object. Also using json_encode now instead of urlencode.